tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/HotChocolate/HotChocolateCommon.cs

```diff
@@ -160,6 +166,122 @@ private static string ConstructErrorMessage(List<IError> executionErrors)
+                    {
+                        eventAttributes.Add(new KeyValuePair<string, object>("path", new[] { path.Name }));
+                    }
+
+                    var code = executionError.Code;
+                    if (code != null)
+                    {
+                        eventAttributes.Add(new KeyValuePair<string, object>("code", code));
+                    }
+
+                    var exception = executionError.Exception;
+                    if (exception != null)
+                    {
+                        eventAttributes.Add(new KeyValuePair<string, object>("stacktrace", exception.StackTrace));
+                    }
+
+                    var extensions = executionError.Extensions;
+                    if (extensions is { Count: > 0 })
+                    {
+                        var configuredExtensions = Tracer.Instance.Settings.GraphQLErrorExtensions;
+
+                        var keys = extensions.Keys.ToList();
+                        foreach (var extension in extensions)
+                        {
+                            if (configuredExtensions.Contains(extension.Key))
+                            {
+                                var key = extension.Key;
+                                var value = extension.Value;
+
+                                if (value == null)
+                                {
+                                    value = "null";
+                                }
+                                else if (value is Array array)
+                                {
+                                    var builder = Util.StringBuilderCache.Acquire();
+
+                                    try
+                                    {
+                                        builder.Append('[');
+                                        for (var k = 0; k < array.Length; k++)
+                                        {
+                                            var item = array.GetValue(k);
+                                            if (k > 0)
+                                            {
+                                                builder.Append(',');
+                                            }
+
+                                            builder.Append(item?.ToString() ?? "null");
+                                        }
+
+                                        builder.Append(']');
+                                        value = Util.StringBuilderCache.GetStringAndRelease(builder);
+                                    }
+                                    catch (Exception ex)
+                                    {
+                                        Console.WriteLine(ex);
+                                        Util.StringBuilderCache.Release(builder);
+                                    }
+                                }
+                                else if (value is ulong or decimal || !Util.SpanEventConverter.IsAllowedType(value))
```

Comment:
nit: The checks for `ulong` and `decimal` seem unnecessary to me, as they are already covered by the `!Util.SpanEventConverter.IsAllowedType(value)` check, right?